test/spec/ServiceManager.ts

```diff
@@ -138,22 +152,58 @@ describe('ServiceManager', () => {
       await client2.serviceManager.stop();
     });
   
-    it('starts autoRenew service after becoming leader (for syncStorage == true)', async () => {
+    it('starts autoRenew service after becoming a leader (for syncStorage == true)', async () => {
       const options = { tokenManager: { syncStorage: true, autoRenew: true } };
       const client = createAuth(options);
+      expect(client.serviceManager.isLeaderRequired()).toBeTruthy();
+      (client.serviceManager.getService('leaderElection') as any)?._becomeLeaderAfterDelay();
+      jest.runAllTimers();
       await client.serviceManager.start();
-      expect(client.serviceManager.isLeader()).toBeFalsy();
-      expect(client.serviceManager.getService('autoRenew')?.isStarted()).toBeFalsy();
-      expect(client.serviceManager.getService('syncStorage')?.isStarted()).toBeTruthy();
-      expect(client.serviceManager.getService('leaderElection')?.isStarted()).toBeTruthy();
-      (client.serviceManager.getService('leaderElection') as any)?._setLeader();
-      await (client.serviceManager.getService('leaderElection') as any)?.onLeader();
       expect(client.serviceManager.isLeader()).toBeTruthy();
       expect(client.serviceManager.getService('autoRenew')?.isStarted()).toBeTruthy();
+      expect(client.serviceManager.getService('syncStorage')?.isStarted()).toBeTruthy();
+      expect(client.serviceManager.getService('leaderElection')?.isStarted()).toBeTruthy();
       await client.serviceManager.stop();
     });
   });
 
+  it('resolves start promise after becoming a leader if any service requires leadership', async () => {
+    const options = { tokenManager: { syncStorage: true, autoRenew: true } };
+    const client = createAuth(options);
+    expect(client.serviceManager.isLeaderRequired()).toBeTruthy();
+    expect(client.serviceManager.hasLeader()).toBeFalsy();
+
+    // leader will be elected in 500ms
+    (client.serviceManager.getService('leaderElection') as any)?._becomeLeaderAfterDelay(500);
+
+    // start service manager, but don't wait to resolve
+    let hasStarted = false;
+    const startPromise = client.serviceManager.start();
+    startPromise.then(() => {
+      hasStarted = true;
+    });
+    
+    // wait until all services are started except autoRenew (waiting for leader)
+    await waitFor(() => (client.serviceManager as any).started);
```

Comment:
serviceManager.start returns a promise, why do we need to loop for a condition to be met?